src/app/_services/persistence.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@angular/core';
+import { Period } from '@app/_models/index';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PersistenceService {
+
+  private db: IDBDatabase = null;
+
+  constructor() { 
+    const request: IDBOpenDBRequest = window.indexedDB.open("PurityDB", 1);
+
+    request.onerror = (event) => { };
+    request.onsuccess = (event: any) => { 
+      this.db = event.target.result;
+    };
+    
+    request.onupgradeneeded = (event: any) => { 
+      this.db = event.target.result;
+
+      const objectStore = this.db.createObjectStore("periods", { 
+        autoIncrement: true 
+      });
+
+      objectStore.createIndex("start", "start", { unique: true });
+
+      objectStore.transaction.oncomplete = (event) => { }
+    }
+  }
+  
+  public addPeriod(period: Period) {
```

Comment:
@telebovich 

1. I see you don't call this method anywhere... Is it for the future purpose?
2. Why you called it addPeriod while in other places it's named `addMachzor`?

Reply:
I started playing with API then I found a wrapper for the API. I will refactor the code a little bit.

Reply:
I merged that, I'll try to find time to play with it better...